src/main/java/com/waverleysoftware/pageobjects/BasePage.java

```diff
@@ -0,0 +1,56 @@
+package com.waverleysoftware.pageobjects;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.List;
+
+import static org.openqa.selenium.support.ui.ExpectedConditions.stalenessOf;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;
+
+/**
+ * A simple base class for processing common methods for all children classes.
+ */
+public class BasePage {
+
+    protected final WebDriver driver;
+    protected final WebDriverWait wait;
+
+    public BasePage(final WebDriver driver) {
+        this.wait = new WebDriverWait(driver, 10);
+        this.driver = driver;
+    }
+
+    public void click(final By locator) {
+        find(locator).click();
+    }
+
+    public void type(final By locator, final String text) {
+        click(locator);
+        if (text != null) {
+            final String existingName = find(locator).getAttribute("value");
+            if (!text.equals(existingName)) {
```

Comment:
if input has value, it doesn't seem to be a valid scenario... we usually type smth and submit... in this case inputs are automatically cleared... if they aren't, most likely there's a bug... so it makes no sense to overhead with additional actions for input retrieval... everything you need is to clear input before typing or not, depending on user's choice...